src/Methods/DebitPaymentMethod.php

```diff
@@ -257,4 +321,63 @@ public function canHandleSubscriptions():bool
     {
         return true;
     }
+
+    /**
+     * @param int $customerId
+     * @return bool
+     */
+    private function isGuest($customerId)
+    {
+        return !$this->accountService->getIsAccountLoggedIn() && (int)$customerId === 0;
```

Comment:
This is not correct. You seem to have problems with boolean arithmetic.
```php
// see old code in isActive()
$isRegular = $this->accountService->getIsAccountLoggedIn() && $contactId > 0;
$isGuest = !$isRegular;
// => $isGuest = !($this->accountService->getIsAccountLoggedIn() && $contactId > 0)
//             = !$this->accountService->getIsAccountLoggedIn() || !($contactId > 0)
//             = !$this->accountService->getIsAccountLoggedIn() || $contactId <= 0
```
So you must change the `&&` by a `||` (we assume `$contactId <= 0` equals `$contactId === 0`).